app/indexers/hyrax/valkyrie_indexer.rb

```diff
@@ -95,7 +95,12 @@ class << self
       # @example
       #     ValkyrieIndexer.for(resource: Book.new) # => #<BookIndexer ...>
       def for(resource:)
-        indexer_class_for(resource).new(resource: resource)
+        case resource
+        when Hyrax::FileSet
+          Hyrax::ValkyrieFileSetIndexer
```

Comment:
This is the 'special resolution logic' to allow `Hyrax::FileSet` to have an indexer that doesn't follow the naming convention.